services/getstream/followTopic.js

```diff
@@ -96,6 +98,22 @@ const addTopicToChatTab = async (
       channel_type: 3
     });
 
+    //remove topic invitations channel
+    if (topicInvitationIds.length > 0) {
+      try {
+        topicInvitationIds.forEach(async (topicInvitationsId) => {
+          const topicInvitationChannel = client.channel(
+            CHANNEL_TYPE_STRING.TOPIC_INVITATION,
+            topicInvitationsId
+          );
+          await topicInvitationChannel.delete();
+        });
+      } catch (error) {
+        console.log('error remove invitations topic channel', error);
+        throw new Error('error remove invitations topic channel');
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a few issues with this block of code:

1. The error message thrown at line 113 is not very descriptive and does not provide enough context about the error. It would be better to include the error message from the caught error.

2. The use of `forEach` with an async function can lead to unhandled promise rejections if any of the promises reject. This is because `forEach` will not wait for promises to resolve before moving on to the next iteration. A better approach would be to use `Promise.all` with `map`.

3. Logging the error to the console (line 112) might expose sensitive information. Consider using a logging library that can obscure sensitive data or remove this log statement.

Here's how you could address these issues:
```diff
- topicInvitationIds.forEach(async (topicInvitationsId) => {
-   const topicInvitationChannel = client.channel(
-     CHANNEL_TYPE_STRING.TOPIC_INVITATION,
-     topicInvitationsId
-   );
-   await topicInvitationChannel.delete();
- });
+ await Promise.all(topicInvitationIds.map(async (topicInvitationsId) => {
+   try {
+     const topicInvitationChannel = client.channel(
+       CHANNEL_TYPE_STRING.TOPIC_INVITATION,
+       topicInvitationsId
+     );
+     await topicInvitationChannel.delete();
+   } catch (error) {
+     throw new Error(`Error removing topic invitation channel: ${error.message}`);
+   }
+ }));
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->